Collector/MigrationsCollector.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MigrationsBundle\Collector;
+
+use Doctrine\Migrations\DependencyFactory;
+use Doctrine\Migrations\Metadata\AvailableMigration;
+use Doctrine\Migrations\Metadata\AvailableMigrationsList;
+use Doctrine\Migrations\Metadata\ExecutedMigration;
+use Doctrine\Migrations\Metadata\ExecutedMigrationsList;
+use Doctrine\Migrations\Metadata\Storage\TableMetadataStorageConfiguration;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+
+class MigrationsCollector extends DataCollector
+{
+    /** @var DependencyFactory */
+    private $dependencyFactory;
+
+    public function __construct(DependencyFactory $dependencyFactory)
+    {
+        $this->dependencyFactory = $dependencyFactory;
+    }
+
+    public function collect(Request $request, Response $response, \Throwable $exception = null)
+    {
+        $metadataStorage = $this->dependencyFactory->getMetadataStorage();
+        $planCalculator = $this->dependencyFactory->getMigrationPlanCalculator();
+        $statusCalculator = $this->dependencyFactory->getMigrationStatusCalculator();
+
+        $this->data['executed_migrations'] = $this->flattenExecutedMigrations($metadataStorage->getExecutedMigrations());
+        $this->data['available_migrations'] = $this->flattenAvailableMigrations($planCalculator->getMigrations());
+        $this->data['new_migrations'] = $this->flattenAvailableMigrations($statusCalculator->getNewMigrations());
+        $this->data['executed_unavailable_migrations'] = $this->flattenExecutedMigrations(
+            $statusCalculator->getExecutedUnavailableMigrations()
+        );
+
+        $configuration = $this->dependencyFactory->getConfiguration()->getMetadataStorageConfiguration();
```

Comment:
if you want to print the storage class, it is more reliable  to use `getMetadataStorage()` (and is never null).

`getMetadataStorageConfiguration` is an optional class to implement since a storage might not be configurable and manage everything internally.

I would still use `getMetadataStorageConfiguration` to print  the table and column name.